content/2015.08.notifications-preamble.md

```diff
@@ -0,0 +1,174 @@
+            Consumer -> Queue [style=dotted];
+        }
+    }
+%}
+
+Pros:
+
+* This model implies a highly limited scope insofar as Kinto is directly concerned; we would basically treat Kinto like a lightweight transport mechanism, and the queue as a fire-and-forget target.
+* The amount of additional code would be relatively small - in essence, only just enough to speak to a queue service and nothing more.
+* We could ship with support for a handful of external queuing services.
+
+Cons:
+
+* We'd have to code and maintain support for a handful of external queuing services.
+* It's ultimately up to the final consumer (receiver, destination, whatever) of the message to deal with the queue. In other words, if the queue is RabbitMQ, then the consumer needs to able to speak RabbitMQ too.
+* Kinto has no control, understanding, or insight into what happens once the message is handed off to the queuing service.
+
+The simplicity of this model, while attractive, has some fatal flaws. Chiefly, it runs contrary to our guideline of tech agnosticism as it would impose a particular external service dependency on the *consumer*. Furthermore, this model flirts with danger in terms of robustness as the fire-and-forget model isn't exactly optimised for reliability (not all queuing mechanisms are better than others, however).
+
+## The Integrated
+
+We then considered what might happen if we were to swing the pendulum directly to the opposite of the spectrum from the previous solution. In this model, we would tightly couple the queuing service directly within Kinto and provide an API that the consumer could subscribe to.
+
+{% graphviz
+    dot {
+        digraph integrated {
+            rankdir=LR;
+            subgraph cluster_internal {
+                style=filled;
+                color=lightgrey;
+                node [style=filled, color=white];
+                Kinto [shape=box];
+                Kinto -> internal_queue [label=1, dir=both];
+                Kinto -> endpoint [label=3];
+            }
+            Queue [shape=box];
+            internal_queue -> Queue [label=2, dir=both];
+            Consumer [shape=box];
+            endpoint -> Consumer [label=4, dir=both];
+        }
+    }
+%}
+Pros:
+
+* Kinto could be shipped as a "turn-key" solution with a fully functioning stack baked right in; while there would be a queuing service involved ([Celery](http://www.celeryproject.org/), for example), it would be completely abstracted by Kinto.
+* Since the queuing mechanism is integral, Kinto would have full visibility and control over processing and delivery.
+* Apart from the queuing service there would be no additional services or software to deploy.
+
+Cons:
+
+* This would involve a fairly serious increase in the scope, complexity, and size of the Kinto code base.
+* The tight coupling means that Kinto, in its entirety, would need to be re-deployed if changes were necessary in the queuing service.
```

Comment:
If the queue service changes, then the queuing mechanism must change as well, which (due to the tight coupling) means a re-deploy.

If I'm interpreting what you've said correctly, you have identified a different problem (which I could add to the list, certainly).